examples/split_plot_rcbd.R

```diff
@@ -0,0 +1,216 @@
+my.data <- read.csv('Lab9ex1.csv')
+#-----------------------------------------------------------------------------#
+
+# The factors in this experiment are recorded with numerical values so they need
+# to be recoded as factors. The user will have to set this manually in the app.
+# NOTE : This is not necessary if `aov()` is used instead of `lm()`.
+#-----------------------------------------------------------------------------#
+my.data$SeedLotA <- as.factor(my.data$SeedLotA)
+my.data$Block <- as.factor(my.data$Block)
+my.data$TrtmtB <- as.factor(my.data$TrtmtB)
+#-----------------------------------------------------------------------------#
+
+# TODO : Is this triple transformation necessary? LOL, I didn't mean it to be a triple transform
+##Testing various transformations for improving assumption tests##
+#(1) Create a sqrt-transformed variable
+my.data$Yield <- sqrt(my.data$Yield)
+
+#(2) Create a log-transformed variable
+my.data$Yield <- log10(my.data$Yield)
+
+#(3)----- Finding the exponent for a power transformation ---- #
+mean.data <- aggregate(Yield ~ SeedLotA + TrtmtB, data = my.data, function(x)
+                       c(logmean=log10(mean(x)), logvar=log10(var(x))))
+power.model <- lm(logvar ~ logmean, data = as.data.frame(mean.data$Yield))
+power <- 1 - summary(power.model)$coefficients[2, 1] / 2
+my.data$Yield <- my.data$Yield^power
+
+# This is the standard model for a split plot RCBD.
+# NOTE : I'm getting this warning message:
+# Warning message:
+# In aov(formula = Yield ~ SeedLotA + Block + Error(SeedLotA:Block) +  :
+#   Error() model is singular
+#-----------------------------------------------------------------------------#
+model <- aov(formula = Yield ~ SeedLotA + Block + Error(SeedLotA:Block) +
+             TrtmtB + SeedLotA:TrtmtB, data = my.data)
+#-----------------------------------------------------------------------------#
+
+# Following the advice in [2], I create the same model without the error term
+# for assumption testing (and post hoc?). The F values and P values are not
+# correct for this model so the ANOVA table should not be shown to the user.
+#-----------------------------------------------------------------------------#
+model.tmp <- aov(formula = Yield ~ Block + SeedLotA + TrtmtB + SeedLotA:Block +
+                 SeedLotA:TrtmtB, data = my.data)
+#-----------------------------------------------------------------------------#
+
+# Plot the four standard fit plots: residuals vs predicted and the Normal Q-Q
+# plot of the residuals.
+dev.new()
+#-----------------------------------------------------------------------------#
+par(mfrow = c(2, 1), oma = c(0, 0, 2, 0))
+plot(model.tmp, which = c(1, 2))
+#-----------------------------------------------------------------------------#
+invisible(dev.copy(png, 'split-plot-rcbd-fit-plots.png'))
+invisible(dev.off())
+
+# Do not do the Shapiro-Wilk Normality Test because it is not straight forward
+# for a split-plot design.
```

Comment:
Let's have a heading that's displayed here that states that: "Shapiro-Wilk Normality Test is not performed because it is not straight forward for a split-plot design."

-I got this from: http://people.stat.sfu.ca/~cschwarz/Stat-650/Notes/PDFbigbook-JMP/JMP-part012.pdf where is says "Normality of errors (residuals). This is a tricky assumption to assess because there are actually two random sources of variation coming from two different normal distributions corresponding to the main-plot and sub-plot experimental units. As before, you have low power to detect non-normality in small samples and in large samples it is not important. Best left to experts. In more advanced classes for graduate level statistics, the best linear unbiased predictors (BLUPs) are found and normal probability plots on the BLUPs are used as diagnostic tools."